packages/venia-concept/src/components/Checkout/braintreeDropin.js

```diff
@@ -91,25 +89,16 @@ const BraintreeDropin = props => {
         return () => {
             didClose = true;
         };
-    }, []);
+    }, [onReady]);
 
     useEffect(() => {
         async function requestPaymentNonce() {
             try {
                 const paymentNonce = await dropinInstance.requestPaymentMethod();
                 onSuccess(paymentNonce);
             } catch (e) {
-                // If payment details were missing or invalid but we have data
-                // from a previous successful submission, use the previous data.
-                const storedPayment = storage.getItem('paymentMethod');
```

Comment:
This was a weird UX. If storage has a valid payment and I enter a new card but _something_ is invalid, it'll just close the form and a user will assume the new info was accepted. I don't think we should do that automatically, so I removed this code. In the future we could display some UX that shows a "edit" button next to saved CC info that would re-mount the braintree dropin but for now we just have to deal with re-entering CC info.

Reply:
Agreed it was a weird UX. If I remember correctly we were trying to avoid having to re-enter CC info if the only thing that was changing was the billing address.

I think you're right that even though re-entering CC info is cumbersome it is better than leaving the user uncertain or unknowledgeable about what CC info is really being used. I'd rather them have to be explicit.